tools/lint/sources.swiftlint.yml

```diff
@@ -25,7 +25,6 @@ only_rules: # we enable lint rules explicitly - only the ones listed below are a
   - empty_xctest_method
   - explicit_top_level_acl
   - first_where
-  - for_where
```

Comment:
I'm disabling this linter rule, because out of suddenly it started raising issues on [this code](https://github.com/DataDog/dd-sdk-ios/blob/a77d0f244e191e3fda6b7280e2d80201ea2dc810/Sources/Datadog/Tracing/DDSpanContext.swift#L22-L23) which is 2 years old:
```swift
for (itemKey, itemValue) in baggageItems.all {
   if callback(itemKey, itemValue) {
```

I think it's because of `for_where` fix in recent SwiftLint [https://github.com/realm/SwiftLint/releases/tag/0.48.0](0.48.0) released last week. My gut feeling is that Bitrise must have upgraded it in their stack (they lock Brew dependencies) and hence we got surprised now 🎁.

Rationale of removal: for 2 years now this syntax was not a problem for us, so IMO it's better to remove this rule, than change our codebase.